### ISSUE TITLE
MacOS version crashes almost instantly.

### ISSUE THREAD

**Issue by Sskenshin24** [PabloMK7/citra#75]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Citra 0c2f076

### Description of Issue

Citra crashes almost instantly when starting full screen or a few seconds after enlarging the window. Happened with both Smash Bros. 3DS and Pokemon Omega Ruby.

Device: Macbook Pro 13 inch M2, 16GB Ram

### Expected Behavior

Citra should crash when enlarging the screen are starting full screen mode.

### Reproduction Steps

Open Citra, start a game, switch to full screen or enlarge the window.

### Log File

[citra_log.txt](https://github.com/PabloMK7/citra/files/14960022/citra_log.txt)

### System Configuration

CPU:M2
GPU/Driver:M2/ Vulkan
RAM:16 GB
OS:MacOS Sonoma 14.4

**Comment by Miguel-hrvs** [comment]
that problem happens in linux too, it seems to be fixed by these GPUCode commits: https://github.com/PabloMK7/citra/pull/27

**Comment by suDisdain** [comment]
Can you re-upload log? Seems to be broken

**Comment by Sskenshin24** [comment]
Thanks. I'm kind of new to this, is there anything I need to do or I just need to wait for an updated build?

**Comment by Miguel-hrvs** [comment]
I think that when this issue happens you get a broken log file

**Comment by Sskenshin24** [comment]
I can try an reupload the log when I get out of work to see if it gets the same issue.

**Comment by suDisdain** [comment]
Omega Ruby may be fixed in #67, was having a similar problem (#66). Wasn't related to fullscreen though. This isn't in latest released build yet.

**Comment by Sskenshin24** [comment]
I just realized I wrote Omega Ruby instead of Alpha Sapphire but I assume it would be the same?

**Comment by suDisdain** [comment]
Yes. Try [this build](https://github.com/PabloMK7/citra/releases/tag/r8433057) for Pokemon. If it doesn't crash, then that issue has been solved already, and Smash Bros is unrelated to it.

**Comment by Miguel-hrvs** [comment]
You're right, pokemon issue is the one that was fixed.

**Comment by Sskenshin24** [comment]
Tried that build and it worked with Pokemon when I click the enlarge button or if I go to view > Fullscreen before I start the game. If the game is already running and I go to view > fullscreen it just shows a black screen. It also worked with Smash Bros 3DS but if I press escape it crashed citra but with pokemon I didn't have that issue.
